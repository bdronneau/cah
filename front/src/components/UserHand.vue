<template>
  <v-container>
    <v-data-table
      v-model="selected"
      :headers="headers"
      :items="cards"
      :single-select=true
      item-key="name"
      show-select
      hide-default-footer
      class="elevation-1"
    >
    </v-data-table>
    <v-btn color="primary" :disabled="selected.length === 0" @click="sendCard">Confirm</v-btn>
    <v-btn text>Reset</v-btn>
  </v-container>
</template>

<script>
export default {
  name: 'UserHand',
  props: {
    cards: Array
  },
  data() {
    return {
      selected: [],
      headers: [
        { text: 'id', value: 'id' },
        { text: 'Name', value: 'name' },
      ],
    }
  },
  methods: {
    sendCard: function() {
      this.$emit("played", this.selected[0]);
    }
  },
};
</script>
