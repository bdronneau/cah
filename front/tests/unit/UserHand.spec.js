// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'

// Components
import UserHand from '@/components/UserHand.vue';

// Utilities
import {
  mount,
  createLocalVue
} from '@vue/test-utils'
Vue.use(Vuetify)

const localVue = createLocalVue()

describe('UserHand.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should have a custom title and match snapshot', () => {
    const wrapper = mount(UserHand, {
      localVue,
      vuetify,
      propsData: {
        cards: [
          {
            id: 1,
            name: "Card 1",
          },
          {
            id: 2,
            name: "Card 2",
          }
        ],
      },
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()
  })
})
