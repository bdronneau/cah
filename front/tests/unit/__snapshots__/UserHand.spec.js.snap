// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHand.vue should have a custom title and match snapshot 1`] = `
<div class="container">
  <div class="v-data-table elevation-1 theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <colgroup>
          <col class="">
          <col class="">
          <col class="">
        </colgroup>
        <thead class="v-data-table-header">
          <tr>
            <th role="columnheader" scope="col" aria-label="" class="text-start" style="width: 1px; min-width: 1px;"><span></span></th>
            <th role="columnheader" scope="col" aria-label="id: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>id</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="Name: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Name</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          </tr>
        </thead>
        <tbody>
          <tr class="">
            <td class="text-start">
              <div class="v-data-table__checkbox v-simple-checkbox">
                <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i>
              </div>
            </td>
            <td class="text-start">1</td>
            <td class="text-start">Card 1</td>
          </tr>
          <tr class="">
            <td class="text-start">
              <div class="v-data-table__checkbox v-simple-checkbox">
                <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i>
              </div>
            </td>
            <td class="text-start">2</td>
            <td class="text-start">Card 2</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div> <button type="button" disabled="disabled" class="v-btn v-btn--contained v-btn--disabled theme--light v-size--default"><span class="v-btn__content">Confirm</span></button> <button type="button" class="v-btn v-btn--flat v-btn--text theme--light v-size--default"><span class="v-btn__content">Reset</span></button>
</div>
`;
